scripts/jenkins/ardana/ansible/init.yml

```diff
@@ -25,6 +25,23 @@
       /usr/sbin/growpart /dev/vda 1 && /sbin/resize2fs /dev/vda1
     ignore_errors: True
 
+  - name: Install the patch command
+    zypper:
+        name: patch
+        state: preset
+    # https://gist.github.com/marcusphi/6791404
+    when: patches is defined and (not patches|trim == '')
+
+  - name: Curl the patches and apply them
+    become: true
+    become_user: ardana
+    shell: |
+        curl -k '{{ item }}' | patch -p1
+    args:
+      chdir: /usr/share/ardana/ansible
```

Comment:
How do you know this is always the right place to apply the patch?  This highlights one of my main concerns about the approach of directly applying a patch to files from installed rpms: it ignores the fact that the `.spec` file has `%build` / `%install` phases which can transform the source files in any way it wants, including moving them to a different location and modifying the contents.  So there is no guarantee that in general the patch will apply cleanly.  Sure, it works in many situations, so I'm really glad that you've shared it with the team; I'm just wondering if it will be good enough for what we need in general.

The approach taken by [`crowbar-testbuild.rb`](https://github.com/SUSE-Cloud/automation/blob/master/scripts/crowbar-testbuild.rb) is safer, because it prepares a patch and then adds that to the `%prep` phase so that it is applied to the sources *before* they are built into the rpm.